Framework assemblies are no longer added to .nuspec when using the new .csproj project system

_Moved from https://github.com/NuGet/Home/issues/4412#issuecomment-287148113:_

_______________

In the marvelous world of `project.json`, specifying a `frameworkAssemblies` node was enough to tell NuGet to add the corresponding entries in the `.nuspec` file:

```json
{
  "frameworks": {
    "net451": {
      "frameworkAssemblies": {
        "System.IdentityModel": "4.0.0.0",
        "System.Xml": "4.0.0.0"
      }
    }
  }
}
```

```xml
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
  <metadata>
    // ...
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="System.IdentityModel" targetFramework=".NETFramework4.5.1" />
      <frameworkAssembly assemblyName="System.Xml" targetFramework=".NETFramework4.5.1" />
    </frameworkAssemblies>
  </metadata>
</package>
```

With .csproj, this is no longer true when using `Reference`:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <ItemGroup>
    <Reference Include="System.IdentityModel" />
    <Reference Include="System.Xml" />
  </ItemGroup>

</Project>
```

This breaking change has obviously important implications, as framework assemblies are no longer brought transitively, which means that the final user of the package has to reference the framework assemblies in his own app.

@rohit21agrawal finally came around to taking your nupkg for a spin. The build fails with the following errors:

```
V:\StackExchange\Dapper\Dapper\Dapper.csproj : error MSB4057: The target "pack" does not exist in the project.
V:\StackExchange\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.csproj : error MSB4057: The target "pack" does not exist in the project.
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(171,15): error MSB4064: The "TfmSpecificAssemblyReferences" parameter is not supported by the "PackTask" task. Verify the parameter exists on the task, and it is a settable public instance property. [V:\StackExchange\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(140,5): error MSB4063: The "PackTask" task could not be initialized with its input parameters.  [V:\StackExchange\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(171,15): error MSB4064: The "TfmSpecificAssemblyReferences" parameter is not supported by the "PackTask" task. Verify the parameter exists on the task, and it is a settable public instance property. [V:\StackExchange\Dapper\Dapper.Contrib\Dapper.Contrib.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(140,5): error MSB4063: The "PackTask" task could not be initialized with its input parameters.  [V:\StackExchange\Dapper\Dapper.Contrib\Dapper.Contrib.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(171,15): error MSB4064: The "TfmSpecificAssemblyReferences" parameter is not supported by the "PackTask" task. Verify the parameter exists on the task, and it is a settable public instance property. [V:\StackExchange\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(140,5): error MSB4063: The "PackTask" task could not be initialized with its input parameters.  [V:\StackExchange\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(171,15): error MSB4064: The "TfmSpecificAssemblyReferences" parameter is not supported by the "PackTask" task. Verify the parameter exists on the task, and it is a settable public instance property. [V:\StackExchange\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(140,5): error MSB4063: The "PackTask" task could not be initialized with its input parameters.  [V:\StackExchange\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(171,15): error MSB4064: The "TfmSpecificAssemblyReferences" parameter is not supported by the "PackTask" task. Verify the parameter exists on the task, and it is a settable public instance property. [V:\StackExchange\Dapper\Dapper\Dapper.StrongName.csproj]
~\.nuget\packages\nuget.build.tasks.pack\4.3.0-beta1-1234\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(140,5): error MSB4063: The "PackTask" task could not be initialized with its input parameters.  [V:\StackExchange\Dapper\Dapper\Dapper.StrongName.csproj]
```

You can try it out for yourself by running `.\build.ps1 -SkipTests`  from this commit: https://github.com/StackExchange/Dapper/commit/972db0625b2a1d0912591bb5a9cf4c62c269f13c

Here's output with all those packages (note it doesn't support net40):
```
PS C:\git\Dapper> .\build.ps1 -SkipTests
Restoring...
  Restoring packages for C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj...
  Restoring packages for C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj...
  Restoring packages for C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj...
  Installing NuGet.Packaging.Core 4.3.0-xprivate-37.
  Installing NuGet.Packaging 4.3.0-xprivate-37.
  Installing NuGet.Versioning 4.3.0-xprivate-37.
  Installing NuGet.Protocol 4.3.0-xprivate-37.
  Installing NuGet.Frameworks 4.3.0-xprivate-37.
  Installing NuGet.LibraryModel 4.3.0-xprivate-37.
  Installing NuGet.Common 4.3.0-xprivate-37.
  Installing NuGet.DependencyResolver.Core 4.3.0-xprivate-37.
  Installing NuGet.Configuration 4.3.0-xprivate-37.
  Installing NuGet.ProjectModel 4.3.0-xprivate-37.
  Installing NuGet.Commands 4.3.0-xprivate-37.
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - net45 (.NETFramework,Version=v4.5)\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - netstandard1.3 (.NETStandard,Version=v1.3) [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : One or more packages are incompatible with .NETFramework,Version=v4.0. [C:\git\Dapper\Dapper.sln]
  Writing lock file to disk. Path: C:\git\Dapper\Dapper.EntityFramework\obj\project.assets.json
  Restore failed in 909.45 ms for C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj.
  Restoring packages for C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj...
  Installing System.Security.Cryptography.ProtectedData 4.0.0.
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - net45 (.NETFramework,Version=v4.5)\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - netstandard1.3 (.NETStandard,Version=v1.3) [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : One or more packages are incompatible with .NETFramework,Version=v4.0. [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - net45 (.NETFramework,Version=v4.5)\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - netstandard1.3 (.NETStandard,Version=v1.3) [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : One or more packages are incompatible with .NETFramework,Version=v4.0. [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - net45 (.NETFramework,Version=v4.5)\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - netstandard1.3 (.NETStandard,Version=v1.3) [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : One or more packages are incompatible with .NETFramework,Version=v4.0. [C:\git\Dapper\Dapper.sln]
  Writing lock file to disk. Path: C:\git\Dapper\Dapper.Rainbow\obj\project.assets.json
  Writing lock file to disk. Path: C:\git\Dapper\Dapper.SqlBuilder\obj\project.assets.json
  Writing lock file to disk. Path: C:\git\Dapper\Dapper.Contrib\obj\project.assets.json
  Restore failed in 1.84 sec for C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj.
  Restore failed in 894.22 ms for C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj.
  Restore failed in 1.85 sec for C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj.
  Restoring packages for C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj...
  Restoring packages for C:\git\Dapper\Dapper.Tests\Dapper.Tests.csproj...
  Restoring packages for C:\git\Dapper\Dapper.Tests.Performance\Dapper.Tests.Performance.csproj...
  Generating MSBuild file C:\git\Dapper\Dapper.Tests.Performance\obj\Dapper.Tests.Performance.csproj.nuget.g.props.
  Writing lock file to disk. Path: C:\git\Dapper\Dapper.Tests.Performance\obj\project.assets.json
  Restore completed in 235.56 ms for C:\git\Dapper\Dapper.Tests.Performance\Dapper.Tests.Performance.csproj.
  Restoring packages for C:\git\Dapper\Dapper\Dapper.StrongName.csproj...
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - net45 (.NETFramework,Version=v4.5)\r [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error :   - netstandard1.3 (.NETStandard,Version=v1.3) [C:\git\Dapper\Dapper.sln]
C:\Program Files\dotnet\sdk\1.0.3\NuGet.targets(97,5): error : One or more packages are incompatible with .NETFramework,Version=v4.0. [C:\git\Dapper\Dapper.sln]
  Writing lock file to disk. Path: C:\git\Dapper\Dapper\obj\project.assets.json
  Restore failed in 378.22 ms for C:\git\Dapper\Dapper\Dapper.StrongName.csproj.
  Restoring packages for C:\git\Dapper\NuGet.FrameworkAssemblyPacker\NuGet.FrameworkAssemblyPacker.csproj...
  Generating MSBuild file C:\git\Dapper\NuGet.FrameworkAssemblyPacker\obj\NuGet.FrameworkAssemblyPacker.csproj.nuget.g.props.
  Writing lock file to disk. Path: C:\git\Dapper\NuGet.FrameworkAssemblyPacker\obj\project.assets.json
  Restore completed in 219.81 ms for C:\git\Dapper\NuGet.FrameworkAssemblyPacker\NuGet.FrameworkAssemblyPacker.csproj.
  Restoring packages for C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj...
  Generating MSBuild file C:\git\Dapper\Dapper.Tests.Contrib\obj\Dapper.Tests.Contrib.csproj.nuget.g.props.
  Writing lock file to disk. Path: C:\git\Dapper\Dapper.Tests.Contrib\obj\project.assets.json
  Restore completed in 991.34 ms for C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj.
  Restoring packages for C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj...
  Generating MSBuild file C:\git\Dapper\Dapper.Tests\obj\Dapper.Tests.csproj.nuget.g.props.
  Writing lock file to disk. Path: C:\git\Dapper\Dapper.Tests\obj\project.assets.json
  Restore completed in 994.85 ms for C:\git\Dapper\Dapper.Tests\Dapper.Tests.csproj.
  Restore completed in 830.46 ms for C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj.
  Restore completed in 795.92 ms for C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj.

  Errors in C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj
      Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:
        - net45 (.NETFramework,Version=v4.5)
        - netstandard1.3 (.NETStandard,Version=v1.3)
      One or more packages are incompatible with .NETFramework,Version=v4.0.

  Errors in C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj
      Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:
        - net45 (.NETFramework,Version=v4.5)
        - netstandard1.3 (.NETStandard,Version=v1.3)
      One or more packages are incompatible with .NETFramework,Version=v4.0.

  Errors in C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj
      Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:
        - net45 (.NETFramework,Version=v4.5)
        - netstandard1.3 (.NETStandard,Version=v1.3)
      One or more packages are incompatible with .NETFramework,Version=v4.0.

  Errors in C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj
      Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:
        - net45 (.NETFramework,Version=v4.5)
        - netstandard1.3 (.NETStandard,Version=v1.3)
      One or more packages are incompatible with .NETFramework,Version=v4.0.

  Errors in C:\git\Dapper\Dapper\Dapper.StrongName.csproj
      Package NuGet.Build.Tasks 4.3.0-xprivate-37 is not compatible with net40 (.NETFramework,Version=v4.0). Package NuGet.Build.Tasks 4.3.0-xprivate-37 supports:
        - net45 (.NETFramework,Version=v4.5)
        - netstandard1.3 (.NETStandard,Version=v1.3)
      One or more packages are incompatible with .NETFramework,Version=v4.0.

  NuGet Config files used:
      C:\git\Dapper\NuGet.Config
      C:\Users\ncraver.STACKEXCHANGE\AppData\Roaming\NuGet\NuGet.Config
      C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

  Feeds used:
      C:\git\Dapper\Nugets
      https://www.nuget.org/api/v2/
      C:\NuGet
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

  Installed:
      11 package(s) to C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj
      12 package(s) to C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj
      12 package(s) to C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj
      12 package(s) to C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj
      11 package(s) to C:\git\Dapper\Dapper.Tests.Performance\Dapper.Tests.Performance.csproj
      12 package(s) to C:\git\Dapper\Dapper\Dapper.StrongName.csproj
      12 package(s) to C:\git\Dapper\NuGet.FrameworkAssemblyPacker\NuGet.FrameworkAssemblyPacker.csproj
      12 package(s) to C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj
      12 package(s) to C:\git\Dapper\Dapper.Tests\Dapper.Tests.csproj
restore failed with exit code 1
At C:\git\Dapper\build.ps1:13 char:5
+     throw "restore failed with exit code $LASTEXITCODE"
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (restore failed with exit code 1:String) [], RuntimeException
    + FullyQualifiedErrorId : restore failed with exit code 1
```

...and here's removing our `net40` targets for the sake of testing:
```
PS C:\git\Dapper> .\build.ps1 -SkipTests
Restoring...
  Restoring packages for C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj...
  Restoring packages for C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj...
  Restoring packages for C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper.EntityFramework\obj\project.assets.json
  Restore completed in 191.51 ms for C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj.
  Restoring packages for C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper.Contrib\obj\project.assets.json
  Restore completed in 553.22 ms for C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj.
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper.SqlBuilder\obj\project.assets.json
  Restore completed in 337.92 ms for C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj.
  Restoring packages for C:\git\Dapper\Dapper.Tests.Performance\Dapper.Tests.Performance.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper.Rainbow\obj\project.assets.json
  Restore completed in 569.59 ms for C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj.
  Restoring packages for C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj...
  Restoring packages for C:\git\Dapper\Dapper.Tests\Dapper.Tests.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper.Tests.Performance\obj\project.assets.json
  Restore completed in 207.76 ms for C:\git\Dapper\Dapper.Tests.Performance\Dapper.Tests.Performance.csproj.
  Restoring packages for C:\git\Dapper\Dapper\Dapper.StrongName.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper\obj\project.assets.json
  Restore completed in 354.56 ms for C:\git\Dapper\Dapper\Dapper.StrongName.csproj.
  Restoring packages for C:\git\Dapper\NuGet.FrameworkAssemblyPacker\NuGet.FrameworkAssemblyPacker.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\NuGet.FrameworkAssemblyPacker\obj\project.assets.json
  Restore completed in 207.01 ms for C:\git\Dapper\NuGet.FrameworkAssemblyPacker\NuGet.FrameworkAssemblyPacker.csproj.
  Restoring packages for C:\git\Dapper\Dapper\Dapper.StrongName.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper.Tests.Contrib\obj\project.assets.json
  Restore completed in 992.63 ms for C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj.
  Restoring packages for C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj...
  Lock file has not changed. Skipping lock file write. Path: C:\git\Dapper\Dapper.Tests\obj\project.assets.json
  Restore completed in 1 sec for C:\git\Dapper\Dapper.Tests\Dapper.Tests.csproj.
  Restore completed in 757.63 ms for C:\git\Dapper\Dapper\Dapper.StrongName.csproj.
  Restore completed in 741.95 ms for C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj.

  NuGet Config files used:
      C:\git\Dapper\NuGet.Config
      C:\Users\ncraver.STACKEXCHANGE\AppData\Roaming\NuGet\NuGet.Config
      C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

  Feeds used:
      C:\git\Dapper\Nugets
      https://www.nuget.org/api/v2/
      C:\NuGet
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
Microsoft (R) Build Engine version 15.1.1012.6693
Copyright (C) Microsoft Corporation. All rights reserved.

  Dapper -> C:\git\Dapper\Dapper\bin\Release\net45\Dapper.dll
  Dapper.StrongName -> C:\git\Dapper\Dapper\bin\Release\net45\Dapper.StrongName.dll
  Dapper -> C:\git\Dapper\Dapper\bin\Release\net451\Dapper.dll
  Dapper.StrongName -> C:\git\Dapper\Dapper\bin\Release\net451\Dapper.StrongName.dll
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper\Dapper.csproj]
  Dapper -> C:\git\Dapper\Dapper\bin\Release\netstandard1.3\Dapper.dll
  Dapper.SqlBuilder -> C:\git\Dapper\Dapper.SqlBuilder\bin\Release\net45\Dapper.SqlBuilder.dll
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper\Dapper.StrongName.csproj]
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj]
  Dapper.SqlBuilder -> C:\git\Dapper\Dapper.SqlBuilder\bin\Release\netstandard1.3\Dapper.SqlBuilder.dll
  Dapper.Rainbow -> C:\git\Dapper\Dapper.Rainbow\bin\Release\net45\Dapper.Rainbow.dll
  Dapper.EntityFramework -> C:\git\Dapper\Dapper.EntityFramework\bin\Release\net45\Dapper.EntityFramework.dll
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj]
  Dapper.Rainbow -> C:\git\Dapper\Dapper.Rainbow\bin\Release\netstandard1.3\Dapper.Rainbow.dll
  Dapper.Contrib -> C:\git\Dapper\Dapper.Contrib\bin\Release\net45\Dapper.Contrib.dll
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj]
  Dapper.Contrib -> C:\git\Dapper\Dapper.Contrib\bin\Release\netstandard1.3\Dapper.Contrib.dll
  NuGet.FrameworkAssemblyPacker -> C:\git\Dapper\NuGet.FrameworkAssemblyPacker\bin\Release\netstandard1.3\NuGet.FrameworkAssemblyPacker.dll
  Dapper.StrongName -> C:\git\Dapper\Dapper\bin\Release\netstandard1.3\Dapper.StrongName.dll
  Dapper.EntityFramework.StrongName -> C:\git\Dapper\Dapper.EntityFramework\bin\Release\net45\Dapper.EntityFramework.StrongName.dll
  Dapper.Tests.Contrib -> C:\git\Dapper\Dapper.Tests.Contrib\bin\Release\netcoreapp1.0\Dapper.Tests.Contrib.dll
  Dapper.Tests -> C:\git\Dapper\Dapper.Tests\bin\Release\net452\Dapper.Tests.dll
  Dapper.Tests -> C:\git\Dapper\Dapper.Tests\bin\Release\netcoreapp1.0\Dapper.Tests.dll
  Dapper.Tests.Performance -> C:\git\Dapper\Dapper.Tests.Performance\bin\Release\net462\Dapper.Tests.Performance.exe

Build succeeded.

C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper\Dapper.csproj]
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper\Dapper.StrongName.csproj]
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj]
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj]
C:\Program Files\dotnet\sdk\1.0.3\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj]
    5 Warning(s)
    0 Error(s)

Time Elapsed 00:00:11.41
Skipping Tests...
Microsoft (R) Build Engine version 15.1.1012.6693
Copyright (C) Microsoft Corporation. All rights reserved.

C:\git\Dapper\Dapper\Dapper.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.Contrib\Dapper.Contrib.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.Tests.Contrib\Dapper.Tests.Contrib.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.SqlBuilder\Dapper.SqlBuilder.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.Rainbow\Dapper.Rainbow.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper\Dapper.StrongName.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.EntityFramework\Dapper.EntityFramework.StrongName.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.Tests.Performance\Dapper.Tests.Performance.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\NuGet.FrameworkAssemblyPacker\NuGet.FrameworkAssemblyPacker.csproj : error MSB4057: The target "pack" does not exist in the project.
C:\git\Dapper\Dapper.Tests\Dapper.Tests.csproj : error MSB4057: The target "pack" does not exist in the project.
```

--- a/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/IPackTaskRequest.cs
+++ b/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/IPackTaskRequest.cs
@@ -16,7 +16,6 @@ namespace NuGet.Build.Tasks.Pack
     public interface IPackTaskRequest<TItem>
     {
         string AssemblyName { get; }
-        TItem[] AssemblyReferences { get; }
         string[] Authors { get; }
         TItem[] BuildOutputInPackage { get; }
         string BuildOutputFolder { get; }
@@ -25,6 +24,7 @@ namespace NuGet.Build.Tasks.Pack
         string Copyright { get; }
         string Description { get; }
         bool DevelopmentDependency { get; }
+        TItem[] FrameworkAssemblyReferences { get; }
         string IconUrl { get; }
         bool IncludeBuildOutput { get; }
         bool IncludeSource { get; }

--- a/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Pack.targets
+++ b/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Pack.targets
@@ -1,4 +1,4 @@
-﻿<!--
+<!--
 ***********************************************************************************************
 NuGet.targets
 
@@ -169,7 +169,7 @@ Copyright (c) .NET Foundation. All rights reserved.
               NoPackageAnalysis="$(NoPackageAnalysis)"
               MinClientVersion="$(MinClientVersion)"
               Serviceable="$(Serviceable)"
-              AssemblyReferences="@(_References)"
+              FrameworkAssemblyReferences="@(_FrameworkAssemblyReferences)"
               ContinuePackingAfterGeneratingNuspec="$(ContinuePackingAfterGeneratingNuspec)"
               NuspecOutputPath="$(BaseIntermediateOutputPath)"
               IncludeBuildOutput="$(IncludeBuildOutput)"
@@ -239,6 +239,26 @@ Copyright (c) .NET Foundation. All rights reserved.
           TaskParameter="TargetOutputs"
           ItemName="_SourceFiles" />
     </MSBuild>
+
+    <MSBuild
+      Projects="$(MSBuildProjectFullPath)"
+      Targets="_GetFrameworkAssemblyReferences"
+      Properties="TargetFramework=%(_TargetFrameworks.Identity);
+                  BuildProjectReferences=false;">
+
+      <Output
+          TaskParameter="TargetOutputs"
+          ItemName="_FrameworkAssemblyReferences" />
+    </MSBuild>
+  </Target>
+
+  <Target Name ="_GetFrameworkAssemblyReferences" DependsOnTargets="ResolveReferences" Returns="@(TfmSpecificFrameworkAssemblyReferences)">
+    <ItemGroup>
+      <TfmSpecificFrameworkAssemblyReferences Include="@(ReferencePath->'%(OriginalItemSpec)')" 
+      Condition="'%(ReferencePath.Pack)' != 'false' AND '%(ReferencePath.ResolvedFrom)' == '{TargetFrameworkDirectory}'">
+        <TargetFramework>$(TargetFramework)</TargetFramework>
+      </TfmSpecificFrameworkAssemblyReferences>
+    </ItemGroup>
   </Target>
   
   <Target Name="_GetBuildOutputFilesWithTfm"

--- a/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTask.cs
+++ b/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTask.cs
@@ -1,4 +1,4 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
@@ -47,7 +47,7 @@ namespace NuGet.Build.Tasks.Pack
         public string NuspecFile { get; set; }
         public string MinClientVersion { get; set; }
         public bool Serviceable { get; set; }
-        public ITaskItem[] AssemblyReferences { get; set; }
+        public ITaskItem[] FrameworkAssemblyReferences { get; set; }
         public bool ContinuePackingAfterGeneratingNuspec { get; set; }
         public string NuspecOutputPath { get; set; }
         public bool IncludeBuildOutput { get; set; }
@@ -120,7 +120,6 @@ namespace NuGet.Build.Tasks.Pack
             return new PackTaskRequest
             {
                 AssemblyName = MSBuildStringUtility.TrimAndGetNullForEmpty(AssemblyName),
-                AssemblyReferences = MSBuildUtility.WrapMSBuildItem(AssemblyReferences),
                 Authors = MSBuildStringUtility.TrimAndExcludeNullOrEmpty(Authors),
                 BuildOutputInPackage = MSBuildUtility.WrapMSBuildItem(BuildOutputInPackage),
                 BuildOutputFolder = MSBuildStringUtility.TrimAndGetNullForEmpty(BuildOutputFolder),
@@ -129,6 +128,7 @@ namespace NuGet.Build.Tasks.Pack
                 Copyright = MSBuildStringUtility.TrimAndGetNullForEmpty(Copyright),
                 Description = MSBuildStringUtility.TrimAndGetNullForEmpty(Description),
                 DevelopmentDependency = DevelopmentDependency,
+                FrameworkAssemblyReferences = MSBuildUtility.WrapMSBuildItem(FrameworkAssemblyReferences),
                 IconUrl = MSBuildStringUtility.TrimAndGetNullForEmpty(IconUrl),
                 IncludeBuildOutput = IncludeBuildOutput,
                 IncludeSource = IncludeSource,
@@ -160,7 +160,7 @@ namespace NuGet.Build.Tasks.Pack
                 Tags = MSBuildStringUtility.TrimAndExcludeNullOrEmpty(Tags),
                 TargetFrameworks = MSBuildStringUtility.TrimAndExcludeNullOrEmpty(TargetFrameworks),
                 TargetPathsToSymbols = MSBuildUtility.WrapMSBuildItem(TargetPathsToSymbols),
-                Title = MSBuildStringUtility.TrimAndGetNullForEmpty(Title),
+                Title = MSBuildStringUtility.TrimAndGetNullForEmpty(Title)
             };
         }
     }

--- a/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs
+++ b/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs
@@ -1,4 +1,4 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
@@ -182,7 +182,35 @@ namespace NuGet.Build.Tasks.Pack
             }
 
             PopulateProjectAndPackageReferences(builder, assetsFile);
+            PopulateFrameworkAssemblyReferences(builder, request);
             return builder;
+        }
+
+        private void PopulateFrameworkAssemblyReferences(PackageBuilder builder, IPackTaskRequest<IMSBuildItem> request)
+        {
+            // First add all the assembly references which are not specific to a certain TFM.
+            var tfmSpecificRefs = new Dictionary<string, IList<string>>(StringComparer.OrdinalIgnoreCase);
+            // Then add the TFM specific framework assembly references, and ignore any which have already been added above.
+            foreach(var tfmRef in request.FrameworkAssemblyReferences)
+            {
+                var targetFramework = tfmRef.GetProperty("TargetFramework");
+
+                if (tfmSpecificRefs.ContainsKey(tfmRef.Identity))
+                {
+                    tfmSpecificRefs[tfmRef.Identity].Add(targetFramework);
+                }
+                else
+                {
+                    tfmSpecificRefs.Add(tfmRef.Identity, new List<string>() { targetFramework });
+                }                
+            }
+
+            builder.FrameworkReferences.AddRange(
+                tfmSpecificRefs.Select(
+                    t => new FrameworkAssemblyReference(
+                        t.Key, t.Value.Select(
+                            k => NuGetFramework.Parse(k))
+                            )));
         }
 
         public PackCommandRunner GetPackCommandRunner(

--- a/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskRequest.cs
+++ b/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskRequest.cs
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using NuGet.Commands;
 using NuGet.Common;
 
@@ -9,7 +10,6 @@ namespace NuGet.Build.Tasks.Pack
     public class PackTaskRequest : IPackTaskRequest<IMSBuildItem>
     {
         public string AssemblyName { get; set; }
-        public IMSBuildItem[] AssemblyReferences { get; set; }
         public string[] Authors { get; set; }
         public IMSBuildItem[] BuildOutputInPackage { get; set; }
         public string BuildOutputFolder { get; set; }
@@ -18,6 +18,7 @@ namespace NuGet.Build.Tasks.Pack
         public string Copyright { get; set; }
         public string Description { get; set; }
         public bool DevelopmentDependency { get; set; }
+        public IMSBuildItem[] FrameworkAssemblyReferences { get; set; }
         public string IconUrl { get; set; }
         public bool IncludeBuildOutput { get; set; }
         public bool IncludeSource { get; set; }

--- a/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs
+++ b/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs
@@ -1,4 +1,4 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
@@ -1625,7 +1625,6 @@ namespace ClassLibrary
             // Arrange
             using (var testDirectory = TestDirectory.Create())
             {
-                //XPlatTestUtils.WaitForDebugger();
                 var projectName = "ClassLibrary1";
                 var workingDirectory = Path.Combine(testDirectory, projectName);
                 
@@ -1680,6 +1679,79 @@ namespace ClassLibrary
                     {
                         Assert.Contains(path, items);
                     }
+                }
+            }
+        }
+
+        [PlatformTheory(Platform.Windows)]
+
+        [InlineData("PresentationFramework",                      true,                           "netstandard1.4;net461",                      "",                             "net461")]
+        [InlineData("PresentationFramework",                      false,                          "netstandard1.4;net461",                      "",                             "net461")]
+        [InlineData("System.IO",                                  true,                           "netstandard1.4;net46",                       "",                             "net46")]
+        [InlineData("System.IO",                                  true,                           "net46;net461",                               "net461",                       "net461")]
+        [InlineData("System.IO",                                  true,                           "net461",                                     "",                             "net461")]
+        public void PackCommand_PackProject_AddsReferenceAsFrameworkAssemblyReference(string referenceAssembly, bool pack,
+            string targetFrameworks, string conditionalFramework, string expectedTargetFramework)
+        {
+            // Arrange
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var projectName = "ClassLibrary1";
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                
+                // Create the subdirectory structure for testing possible source paths for the content file
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+
+                msbuildFixture.CreateDotnetNewProject(testDirectory.Path, projectName, " classlib");
+
+                using (var stream = new FileStream(projectFile, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+                    var frameworkProperty = "TargetFrameworks";
+                    if(targetFrameworks.Split(';').Count() == 1)
+                    {
+                        frameworkProperty = "TargetFramework";
+                    }
+                    ProjectFileUtils.SetTargetFrameworkForProject(xml, frameworkProperty, targetFrameworks);
+
+                    var attributes = new Dictionary<string, string>();
+
+                    var properties = new Dictionary<string, string>();
+                    if (!pack)
+                    {
+                        attributes["Pack"] = "false";
+                    }
+                    ProjectFileUtils.AddItem(
+                        xml,
+                        "Reference",
+                        referenceAssembly,
+                        conditionalFramework,
+                        properties,
+                        attributes);
+
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                msbuildFixture.RestoreProject(workingDirectory, projectName, string.Empty);
+                var nupkgPath = Path.Combine(workingDirectory, $"{projectName}.1.0.0.nupkg");
+                var nuspecPath = Path.Combine(workingDirectory, "obj", $"{projectName}.1.0.0.nuspec");
+                // Act
+                msbuildFixture.PackProject(workingDirectory, projectName, $"-o {workingDirectory}");
+
+                // Assert
+                Assert.True(File.Exists(nupkgPath), "The output .nupkg is not in the expected place");
+                Assert.True(File.Exists(nuspecPath), "The intermediate nuspec file is not in the expected place");
+                //XPlatTestUtils.WaitForDebugger();
+                using (var nupkgReader = new PackageArchiveReader(nupkgPath))
+                {
+                    var expectedFrameworks = expectedTargetFramework.Split(';');
+                    var frameworkItems = nupkgReader.GetFrameworkItems();
+                    foreach(var framework in expectedFrameworks)
+                    {
+                        var nugetFramework = NuGetFramework.Parse(framework);
+                        var frameworkSpecificGroup = frameworkItems.Where(t => t.TargetFramework.Equals(nugetFramework)).First();
+                        Assert.True(frameworkSpecificGroup.Items.Contains(referenceAssembly) == pack);
+                    }                    
                 }
             }
         }

--- a/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskLogicTests.cs
+++ b/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskLogicTests.cs
@@ -406,7 +406,8 @@ namespace NuGet.Build.Tasks.Pack.Test
                         {"FinalOutputPath", dllPath },
                         {"TargetFramework", "net45" }
                     })},
-                    Logger = new TestLogger()
+                    Logger = new TestLogger(),
+                    FrameworkAssemblyReferences = new MSBuildItem[]{}
                 };
             }
 

--- a/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskTests.cs
+++ b/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskTests.cs
@@ -203,7 +203,7 @@ namespace NuGet.Build.Tasks.Pack.Test
             // Arrange
             var target = new PackTask
             {
-                AssemblyReferences = new[] { null, new Mock<ITaskItem>().Object },
+                FrameworkAssemblyReferences = new[] { null, new Mock<ITaskItem>().Object },
                 PackageFiles = new[] { null, new Mock<ITaskItem>().Object },
                 PackageFilesToExclude = new[] { null, new Mock<ITaskItem>().Object },
                 PackItem = new Mock<ITaskItem>().Object,
@@ -214,7 +214,7 @@ namespace NuGet.Build.Tasks.Pack.Test
             var actual = GetRequest(target);
 
             // Assert
-            Assert.Equal(1, actual.AssemblyReferences.OfType<MSBuildTaskItem>().Count());
+            Assert.Equal(1, actual.FrameworkAssemblyReferences.OfType<MSBuildTaskItem>().Count());
             Assert.Equal(1, actual.PackageFiles.OfType<MSBuildTaskItem>().Count());
             Assert.Equal(1, actual.PackageFilesToExclude.OfType<MSBuildTaskItem>().Count());
             Assert.NotNull(actual.PackItem);
@@ -227,7 +227,7 @@ namespace NuGet.Build.Tasks.Pack.Test
             // Arrange
             var target = new PackTask
             {
-                AssemblyReferences = null,
+                FrameworkAssemblyReferences = null,
                 Authors = null,
                 PackageFiles = null,
                 PackageFilesToExclude = null,
@@ -243,7 +243,7 @@ namespace NuGet.Build.Tasks.Pack.Test
             var actual = GetRequest(target);
 
             // Assert
-            Assert.Equal(0, actual.AssemblyReferences.Length);
+            Assert.Equal(0, actual.FrameworkAssemblyReferences.Length);
             Assert.Equal(0, actual.Authors.Length);
             Assert.Equal(0, actual.PackageFiles.Length);
             Assert.Equal(0, actual.PackageFilesToExclude.Length);
@@ -262,7 +262,7 @@ namespace NuGet.Build.Tasks.Pack.Test
             var target = new PackTask
             {
                 AssemblyName = "AssemblyName",
-                AssemblyReferences = new ITaskItem[0],
+                FrameworkAssemblyReferences = new ITaskItem[0],
                 Authors = new string[0],
                 BuildOutputFolder = "BuildOutputFolder",
                 ContentTargetFolders = new string[] { "ContentTargetFolders" } ,

